Lap01_1.1.3_Nguyen Thi Thu Thao
Red Wine Quality_EDA & Classification

In [1]:
import numpy as np 
import pandas as pd
from scipy import stats

In [5]:
wine_data = pd.read_csv("winequality-red.csv") 

In [6]:
wine_data = wine_data[
    ["alcohol", "volatile acidity", "sulphates", "pH", "quality"]
]

In [7]:
print(wine_data.head())

   alcohol  volatile acidity  sulphates    pH  quality
0      9.4              0.70       0.56  3.51        5
1      9.8              0.88       0.68  3.20        5
2      9.8              0.76       0.65  3.26        5
3      9.8              0.28       0.58  3.16        6
4      9.4              0.70       0.56  3.51        5


In [8]:
wine_data.dtypes 
wine_data.shape 

(1599, 5)

In [2]:
target_column = "alcohol"

In [9]:
# 1. Tính giá trị trung bình (Mean)
data_mean = np.mean(wine_data[target_column])

In [11]:
# 2. Tính giá trị trung vị (Median)
data_median = np.median(wine_data[target_column])
# 3. Tính giá trị yếu vị / yếu tố xuất hiện nhiều nhất (Mode)
data_mode = stats.mode(wine_data[target_column], keepdims=True).mode[0]

# 4. Tính phương sai (Variance)
data_variance = np.var(wine_data[target_column])

# 5. Tính độ lệch chuẩn (Standard Deviation)
data_sd = np.std(wine_data[target_column])

# 6. Tính giá trị lớn nhất (Max) và nhỏ nhất (Min)
data_max = np.max(wine_data[target_column])
data_min = np.min(wine_data[target_column])

# 7. Tính bách phân vị thứ 60 (60th percentile)
data_percentile = np.percentile(wine_data[target_column], 60)

# 8. Tính tứ phân vị thứ 3 / mức 75% (Quartiles)
data_quartile = np.quantile(wine_data[target_column], 0.75)

# 9. Tính khoảng biến thiên tứ phân vị (IQR)
data_IQR = stats.iqr(wine_data[target_column])

In [12]:
# --- In kết quả ra màn hình kiểm tra ---
print(f"--- Thống kê mô tả cho cột [{target_column}] ---")
print(f"Mean (Trung bình): {data_mean:.2f}")
print(f"Median (Trung vị): {data_median:.2f}")
print(f"Mode (Yếu vị): {data_mode}")
print(f"Variance (Phương sai): {data_variance:.2f}")
print(f"Standard Deviation (Độ lệch chuẩn): {data_sd:.2f}")
print(f"Min - Max (Nhỏ nhất - Lớn nhất): {data_min} - {data_max}")
print(f"60th Percentile: {data_percentile:.2f}")
print(f"75th Quantile (Tứ phân vị Q3): {data_quartile:.2f}")
print(f"IQR (Khoảng biến thiên tứ phân vị): {data_IQR:.2f}")

--- Thống kê mô tả cho cột [alcohol] ---
Mean (Trung bình): 10.42
Median (Trung vị): 10.20
Mode (Yếu vị): 9.5
Variance (Phương sai): 1.13
Standard Deviation (Độ lệch chuẩn): 1.07
Min - Max (Nhỏ nhất - Lớn nhất): 8.4 - 14.9
60th Percentile: 10.50
75th Quantile (Tứ phân vị Q3): 11.10
IQR (Khoảng biến thiên tứ phân vị): 1.60


Hương vị chủ đạo: Tập dữ liệu này phản ánh một bộ sưu tập rượu đỏ thuộc phân khúc vừa và nhẹ (Light to Medium-bodied) là chủ yếu, do phần lớn dữ liệu tích tụ dưới mức 11%. Những chai rượu có độ cồn vượt trội (13% - 14.9%) chiếm tỷ lệ khá ít.

Mối liên hệ với Chất lượng (quality): Như chúng ta đã biết ở ma trận tương quan trước đó, alcohol có tương quan thuận mạnh nhất với chất lượng rượu. Việc nồng độ cồn tập trung nhiều ở mức thấp (9.5% - 10.2%) giải thích lý do tại sao ở biến quality, số lượng lớn các chai rượu chỉ đạt điểm trung bình (điểm 5 và 6). Những chai đạt điểm xuất sắc (7 và 8) nhiều khả năng sẽ rơi vào nhóm 25% có nồng độ cồn cao nhất (từ 11.1% trở lên).

Xử lý dữ liệu: Vì biến alcohol có phân bổ lệch phải, nếu bạn định sử dụng các thuật toán nhạy cảm với hình dáng phân bổ (như Linear/Logistic Regression), bạn có thể cân nhắc áp dụng kỹ thuật biến đổi Logarit (Log Transformation) để đưa dữ liệu về dạng phân bổ chuẩn hơn. Nếu dùng các mô hình dạng cây (Decision Tree, Random Forest, XGBoost), bạn có thể sử dụng trực tiếp các con số này mà không cần biến đổi.